crypto-attributes/src/entity_derive.rs

```diff
@@ -0,0 +1,567 @@
+            #entity_generic_impl
+            #entity_wasm_impl
+            #entity_transaction_ext_impl
+        }
+    }
+
+    fn entity_base_impl(&self) -> proc_macro2::TokenStream {
+        let Self {
+            collection_name,
+            struct_name,
+            ..
+        } = self;
+
+        // Identical for both wasm and non-wasm
+        quote! {
+            #[cfg_attr(target_family = "wasm", async_trait::async_trait(?Send))]
+            #[cfg_attr(not(target_family = "wasm"), async_trait::async_trait)]
+            impl crate::entities::EntityBase for #struct_name {
+                type ConnectionType = crate::connection::KeystoreDatabaseConnection;
+                type AutoGeneratedFields = ();
+                const COLLECTION_NAME: &'static str = #collection_name;
+
+                fn to_missing_key_err_kind() -> crate::MissingKeyErrorKind {
+                    crate::MissingKeyErrorKind::#struct_name
+                }
+
+                fn to_transaction_entity(self) -> crate::transaction::dynamic_dispatch::Entity {
+                    crate::transaction::dynamic_dispatch::Entity::#struct_name(self)
+                }
+            }
+        }
+    }
+
+    fn entity_generic_impl(&self) -> proc_macro2::TokenStream {
+        let Self {
+            collection_name,
+            struct_name,
+            id,
+            id_type,
+            id_name,
+            blob_columns,
+            blob_column_names,
+            all_columns,
+            ..
+        } = self;
+
+        let string_id_conversion = matches!(id_type, ColumnType::String).then(|| {
+            quote! { let #id: String = id.try_into()?; }
+        });
+
+        let id_getter = match id_type {
+            ColumnType::String => quote! {self.#id.as_bytes() },
+            ColumnType::Bytes => quote! { &self.#id[..] },
+        };
+
+        let id_field_construct_self = match id_type {
+            ColumnType::String => quote! { #id, },
+            ColumnType::Bytes => quote! { #id: id.to_bytes(), },
+        };
+
+        let id_slice = match id_type {
+            ColumnType::String => quote! { #id.as_str() },
+            ColumnType::Bytes => quote! { #id.as_slice() },
+        };
+
+        quote! {
+            #[cfg(not(target_family = "wasm"))]
+            #[async_trait::async_trait]
+            impl crate::entities::Entity for #struct_name {
+                fn id_raw(&self) -> &[u8] {
+                    #id_getter
+                }
+
+                async fn find_all(
+                    conn: &mut Self::ConnectionType,
+                    params: crate::entities::EntityFindParams,
+                ) -> crate::CryptoKeystoreResult<Vec<Self>> {
+                    let transaction = conn.transaction()?;
+                    let query: String = format!("SELECT rowid, {} FROM {} {}", #id_name, #collection_name, params.to_sql());
+
+                    let mut stmt = transaction.prepare_cached(&query)?;
```

Comment:
Definitely a code smell to directly inject values into a SQL string and then execute it, though I suppose we can't really help it in the case of `params` if `to_sql` produces a `WHERE` clause. 

It's particularly odd here though because we have the option to bake `#id_name` and `#collection_name` into the query directly.